turbinia/evidence.py

```diff
@@ -958,3 +958,35 @@ def _postprocess(self):
       # Unmount the container's filesystem
       mount_local.PostprocessUnmountPath(self._container_fs_path)
       self.state[EvidenceState.CONTAINER_MOUNTED] = False
+
+
+class ExpertWitnessFormat(RawDisk):
+  """Evidence object for a EWF based evidence.
+
+  Attributes:
+    device_path (str): Path to the mounted loop device.
+	ewf_path (str): Path to mounted EWF image.
+	mount_path (str): Path to EWF mount directory.
+  """
+  POSSIBLE_STATES = [EvidenceState.ATTACHED] 
+
+  def __init__(self, *args, **kwargs):
+    """Initialization for EWF evidence object."""
+    self.device_path = None
+    self.ewf_path = None
+    self.mount_path = None
+    super(ExpertWitnessFormat, self).__init__(*args, **kwargs)
+
+  def _preprocess(self, _, required_states):
+      if EvidenceState.ATTACHED in required_states:
+          self.mount_path = mount_local.PreprocessMountEwfDisk(self.source_path)
+          self.ewf_path = self.mount_path + "/ewf1"
```

Comment:
It's been a while since I've used ewfmount, and I don't have a good test E01 to test it on, but is this is the path to the mounted image or the block device?  If it's not mounted already, maybe we can add that into the preprocessor as well.   If it is mounted, I don't think the PreprocessLosetup() method will work (or would be needed), and we could potentially also set the `self.state[EvidenceState.MOUNTED] == True`

Reply:
there is an ewf test file in dfvfs test_data

Reply:
Changed mount_path to block_path as it represents the directory of the ewf device and added self.state[EvidenceState.MOUNTED] = True after mounting the ewf. Does it look good?